frontend/src/routes/(authenticated)/project/create/+page.svelte

```diff
@@ -205,14 +203,13 @@
     );
   });
 
+  const calculatedCode = $derived(buildProjectCode($form.languageCode, $form.type, $form.retentionPolicy));
+  // TODO: This causes an infinite loop when converted to $effect. Need to do something clever with superforms to avoid the infinite loop.
```

Comment:
Now that you've added `runed`, I think this might be a good way to solve it.
```suggestion
const isCustomCode = $derived(() => $form.customCode);
// I'm not entirely sure if we need isCustomCode or if we can watch $form.customCode, but I'm assuming it's the former.
watch(() => isCustomCode, () => {
  if (!$form.customCode) {
    form.update(
        (form) => {
          form.code = calculatedCode;
          return form;
        },
        { taint: false },
      );
  }
});
```

Reply:
I've learned that we will also need to untrack `$form` access when it's not a direct write. Otherwise this, or something along those lines, should work.

Reply:
Fixed in commit a362333b.

Reply:
By untracking the form store, I don't think this efect was tracking anything anymore.
So, I did some refactoring to restore the reactivity.